### PR TITLE
Issue 8 update to new api

### PR DESCRIPTION
**Pull request by ShaneKilkelly**
Addresses #8

### CODE REVIEW

File: doc_src/Frontpage.md [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
 
-To get started, read the [Hammer Tutorial](https://hexdocs.pm/hammer/tutorial.html) first, then add the `hammer_backend_redis` dependency:
+To get started, read
+the [Hammer Tutorial](https://hexdocs.pm/hammer/tutorial.html) first, then add
+the `hammer_backend_redis` dependency:
 
 ```elixir
     def deps do
-      [{:hammer_backend_redis, "~> 0.1.0"}]
+      [{:hammer_backend_redis, "~> 1.0.0"}]
```
Comment (sobolevn) [inline doc_src/Frontpage.md]: I generally don't put the patch version to the readme. Just `"~> 1.0"`

File: README.md [hunk 2]
```diff
@@ -22,12 +22,22 @@ end
 
 ## Usage
 
+Just configure the `:hammer` application to use the Redis backend:
```
Comment (sobolevn) [inline README.md]: I would you mind removing "Just"?

File: doc_src/Frontpage.md [hunk 3]
```diff
@@ -1,28 +1,35 @@
...
     end
 ```
 
-... then start a `Hammer.Backend.Redis` process:
+... then configure the `:hammer` application to use the Redis backend:
 
 ```elixir
-      worker(Hammer.Backend.Redis, [[expiry_ms: 1000 * 60 * 2,
-                                     redix_config: [host: "localhost", port: 6379]]]),
+config :hammer,
```
Comment (sobolevn) [inline doc_src/Frontpage.md]: It is the same as in "Usage" section. Maybe we can provide another example? Or remove it.

File: lib/hammer/backend/redis/supervisor.ex [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+  def start_link(config, opts) do
+    Supervisor.start_link(__MODULE__, config, opts)
+  end
+
+  def init(config) do
+    redix_config = Keyword.get(
+      config,
+      :redix_config,
+      Keyword.get(config, :redis_config, [])
```
Comment (sobolevn) [inline lib/hammer/backend/redis/supervisor.ex]: I would add a comment here. That does require some explanation.

File: mix.exs [hunk 5]
```diff
@@ -29,7 +29,7 @@ defmodule HammerBackendRedis.Mixfile do
   # Type "mix help compile.app" for more information
   def application do
     # Specify extra applications you'll use from Erlang/Elixir
-    [extra_applications: [:logger]]
+    [extra_applications: [:logger, :redix]]
```
Comment (sobolevn) [inline mix.exs]: No need to put apps from `deps` to `extra_applications` if I am correct.

**Comment by sobolevn** [comment]
@ShaneKilkelly great work!

File: README.md (hunk 2 above)
Comment (ShaneKilkelly) [reply README.md]: Yeah, 'Just' is a bit too informal, will rephrase

**Comment by ShaneKilkelly** [comment]
@sobolevn, updated :)

**Pull request merged**